spring-cloud-zookeeper-core/src/main/java/org/springframework/cloud/zookeeper/ZookeeperAutoConfiguration.java

```diff
@@ -53,21 +53,28 @@ public ZookeeperProperties zookeeperProperties() {
 
 	@Bean(destroyMethod = "close")
 	@ConditionalOnMissingBean
-	public CuratorFramework curatorFramework(
-			RetryPolicy retryPolicy,
+	public CuratorFramework curatorFramework(RetryPolicy retryPolicy,
 			ZookeeperProperties properties,
+			ObjectProvider<CuratorFrameworkCustomizer> optionalCuratorFrameworkCustomizerProvider,
 			ObjectProvider<EnsembleProvider> optionalEnsembleProvider,
-			ObjectProvider<TracerDriver> optionalTracerDriverProvider)
-	throws Exception {
+			ObjectProvider<TracerDriver> optionalTracerDriverProvider) throws Exception {
 		CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+
 		EnsembleProvider ensembleProvider = optionalEnsembleProvider.getIfAvailable();
 		if (ensembleProvider != null) {
 			builder.ensembleProvider(ensembleProvider);
 		}
 		else {
 			builder.connectString(properties.getConnectString());
 		}
-		CuratorFramework curator = builder.retryPolicy(retryPolicy).build();
+		builder.sessionTimeoutMs((int) properties.getSessionTimeout().toMillis())
+				.connectionTimeoutMs((int) properties.getConnectionTimeout().toMillis())
+				.retryPolicy(retryPolicy);
+
+		optionalCuratorFrameworkCustomizerProvider
+				.ifAvailable(customizer -> customizer.customize(builder));
```

Comment:
so this should still work on a list. Hence when I said before `optionalCuratorFrameworkCustomizerProvider.orderdStream().forEach()`...

Reply:
Sorry I missed that. Updated the code to reflect the suggestion as well as the test to ensure the call order.